test/e2e/network_policy.go

```diff
@@ -0,0 +1,438 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/util/intstr"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+/*
+These Network Policy tests create two services A and B.
```

Comment:
My understanding is that these tests exercise the following features:

- DefaulyDeny support - No traffic is allowed.
- Ports - All TCP port X is allowed on all pods in a given namespace.

Could we also add tests for these?
- Intra-namespace policy.  E.g TCP port X is allowed from a subset of pods in a given namespace using podSelector.
- Namespace selection.  E.g Traffic is allowed from all pods in target namespace using namespaceSelector, but not namespaces which don't match the selector.

This will at least cover all the main primitives within NetworkPolicy (ports, pod selectors, namespace selectors).

Later, we can add tests for things like:
- Overlapping podSelectors
- More than one NetworkPolicyIngressRule per policy.
- UDP
- etc.